pandas/core/arrays/_mixins.py

```diff
@@ -204,10 +205,20 @@ def __setitem__(self, key, value):
     def _validate_setitem_value(self, value):
         return value
 
+    @overload
+    def __getitem__(self, item: int | np.integer) -> Any:
```

Comment:
item or key? item would be consistent with base class, key would be consistent with non-overload signature.

Reply:
fixed in next commit